src/CollectionsAndGenerics/Program.cs

```diff
@@ -0,0 +1,96 @@
+﻿// <copyright file="Program.cs" company="PlaceholderCompany">
+// Copyright (c) PlaceholderCompany. All rights reserved.
+// </copyright>
+
+namespace CollectionsAndGenerics
+{
+    using System.Text.RegularExpressions;
+
+    /// <summary>
+    /// Program class - Asks for user choice of operation to perform with collections
+    /// </summary>
+    public partial class Program
+    {
+        /// <summary>
+        /// CollectionsAvailable - All collections provided to the user
+        /// </summary>
+        private enum CollectionsAvailable
+        {
+            Lists = 1,
+            Stacks = 2,
+            Queues = 3,
+            Dictionary = 4,
+            Exit = 5,
+        }
+
+        /// <summary>
+        /// Main function - Entry point and starts the application
+        /// </summary>
+        /// <param name="args">String array of arguments</param>
+        public static void Main(string[] args)
+        {
+            bool exit = false;
+            do
+            {
+                Console.WriteLine("COLLECTIONS");
+                foreach (CollectionsAvailable collections in Enum.GetValues(typeof(CollectionsAvailable)))
+                {
+                    Console.WriteLine($"{(int)collections}. {collections.ToString()}");
+                }
+
+                Console.WriteLine("Specify the type of collection you are interested to work with(Eg 1 for Lists): ");
+                int userChoice = ValidateInput.Program.GetValidUserChoice(Enum.GetNames(typeof(CollectionsAvailable)).Length);
```

Comment:
Can we use extension method instead of just having the GetValidUserChoice as static?

```suggestion
               Enum.GetNames(typeof(CollectionsAvailable)).Length.GetValidUserChoice();
```

I have renamed the Program.cs as `ValidateUserInput` to avoid confusions for this comment screenshot below
![image](https://github.com/Subhasri-V-P/CSharp/assets/97607877/99863887-495c-4aee-9f5a-a2eda2b0da65)

![image](https://github.com/Subhasri-V-P/CSharp/assets/97607877/4591e25d-63a0-47dc-8f6d-789bcf3bfdd0)

These are the changes you have to do for the rest of the methods as well, 

I also wanted you to explore on extension methods

Reply:
Changes made in ValidateInput project

Reply:
> Can we use extension method instead of just having the GetValidUserChoice as static?
> 
> I have renamed the Program.cs as `ValidateUserInput` to avoid confusions for this comment screenshot below ![image](https://user-images.githubusercontent.com/97607877/271993480-99863887-495c-4aee-9f5a-a2eda2b0da65.png)
> 
> ![image](https://user-images.githubusercontent.com/97607877/271993691-4591e25d-63a0-47dc-8f6d-789bcf3bfdd0.png)
> 
> These are the changes you have to do for the rest of the methods as well,
> 
> I also wanted you to explore on extension methods

I still could see Program class have not been renamed
![image](https://github.com/Subhasri-V-P/CSharp/assets/97607877/e2552444-cae2-4094-91eb-6bcfc6f4f1e6)

Reply:
I thought, it was mentioned that this class's name was changed to avoid confusion for me to understand where I should change the methods to extension methods.

Reply:
I have refactored and updated now